src/explore-education-statistics-common/src/validation/serverValidations.ts

```diff
@@ -163,13 +163,22 @@ export function isServerValidationError(
 
 export function hasServerValidationError(
   error: Error,
-  errorMessage: string,
+  ...errorCodes: string[]
 ): boolean {
   if (isServerValidationError(error) && error.response?.data) {
-    const errors = Object.values(error.response?.data.errors);
-    if (errors.flat().includes(errorMessage)) {
-      return true;
-    }
+    const existingCodes = Object.values(error.response?.data.errors).flat();
+    return existingCodes.every(existingCode =>
+      errorCodes.includes(existingCode),
+    );
   }
   return false;
 }
+
+export function getServerValidationError(error: Error): string | null {
+  if (!isServerValidationError(error) || !error.response?.data) {
+    return null;
+  }
+
+  const errors = Object.values(error.response?.data.errors);
+  return errors.flat()[0];
```

Comment:
This doesn't seem very safe - for example, we could pass in a server validation error where there are field errors, but no server error. It's errors look like:

```ts
errors: {
  theField: ["Message"]
}
```

With the above example and the current logic, we'd end up returning `Message`, even though it's not correct.

We should probably allow this method to specify the field name that we're looking for an error message under so that we simply need to do this:

```ts
export function getServerValidationErrorMessage<T extends string = string>(
  error: AxiosError<ServerValidationErrorResponse>, 
  fieldName = ''
): T | undefined {
  return error.response.data.errors[fieldName]?.[0];
}
```

We default to a `fieldName = ''` as this is the key corresponding to global errors.

Reply:
Cool, have done ta.